azurerm/internal/services/mssql/resource_arm_mssql_virtual_machine.go

```diff
@@ -0,0 +1,425 @@
+							Required: true,
+							ValidateFunc: validation.StringInSlice([]string{
+								string(sqlvirtualmachine.Monday),
+								string(sqlvirtualmachine.Tuesday),
+								string(sqlvirtualmachine.Wednesday),
+								string(sqlvirtualmachine.Thursday),
+								string(sqlvirtualmachine.Friday),
+								string(sqlvirtualmachine.Saturday),
+								string(sqlvirtualmachine.Sunday),
+							}, false),
+						},
+
+						"maintenance_window_duration_in_minutes": {
+							Type:         schema.TypeInt,
+							Required:     true,
+							ValidateFunc: validation.IntBetween(30, 180),
+						},
+
+						"maintenance_window_starting_hour": {
+							Type:         schema.TypeInt,
+							Required:     true,
+							ValidateFunc: validation.IntBetween(0, 23),
+						},
+					},
+				},
+			},
+
+			"key_vault_credential": {
+				Type:     schema.TypeList,
+				Optional: true,
+				MaxItems: 1,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"name": {
+							Type:         schema.TypeString,
+							Required:     true,
+							ValidateFunc: validation.StringIsNotEmpty,
+							//api will add updated credential name, and return "sqlvmName:name1,sqlvmName:name2"
+							DiffSuppressFunc: mssqlVMCredentialNameDiffSuppressFunc,
+						},
+
+						"key_vault_url": {
+							Type:         schema.TypeString,
+							Required:     true,
+							ForceNew:     true,
+							Sensitive:    true,
+							ValidateFunc: validation.IsURLWithHTTPS,
+						},
+
+						"service_principal_name": {
+							Type:         schema.TypeString,
+							Required:     true,
+							ForceNew:     true,
+							Sensitive:    true,
+							ValidateFunc: validation.StringIsNotEmpty,
+						},
+
+						"service_principal_secret": {
+							Type:         schema.TypeString,
+							Required:     true,
+							ForceNew:     true,
+							Sensitive:    true,
+							ValidateFunc: validation.StringIsNotEmpty,
+						},
+					},
+				},
+			},
+
+			"r_services_enabled": {
+				Type:     schema.TypeBool,
+				Optional: true,
+			},
+
+			"sql_connectivity_port": {
+				Type:         schema.TypeInt,
+				Optional:     true,
+				Default:      1433,
+				ValidateFunc: validation.IntBetween(1024, 65535),
+			},
+
+			"sql_connectivity_type": {
+				Type:     schema.TypeString,
+				Optional: true,
+				Default:  string(sqlvirtualmachine.PRIVATE),
+				ValidateFunc: validation.StringInSlice([]string{
+					string(sqlvirtualmachine.LOCAL),
+					string(sqlvirtualmachine.PRIVATE),
+					string(sqlvirtualmachine.PUBLIC),
+				}, false),
+			},
+
+			"sql_connectivity_update_password": {
+				Type:         schema.TypeString,
+				Optional:     true,
+				Sensitive:    true,
+				ValidateFunc: validation.StringIsNotEmpty,
+			},
+
+			"sql_connectivity_update_username": {
+				Type:         schema.TypeString,
+				Optional:     true,
+				Sensitive:    true,
+				ValidateFunc: validate.MsSqlVMLoginUserName,
+			},
+
+			"sql_virtual_machine_group_id": {
```

Comment:
given we can't set this yet due to no AADS server resource could we just remove it?